### PR TITLE
Mana Steal Nerf

### LINKED ISSUE #93: Manasteal Nerf
I hope I don't need to elaborate on this one

### PR DESCRIPTION
**Pull request by SoulMuncher**
Closes #93
- Mana Steal requires Mana Burn
- Mana Steal now steals 60 mana per stack, split between all targeted enemies

QA'd sp, mp

### CODE REVIEW

**Comment by jdoleary** [comment]
I think this is the solution
https://discord.com/channels/1032294536640200766/1032294537235812404/1185003944540389446
Bigger cap, but global cap.  Stacking increases the cap.  This way the max mana is capped regardless of how many units are targeted. Brilliant idea by Coul.

What do you think?

**Comment by jdoleary** [comment]
Continuing our discussion about Coul's mana steal idea.
The reason I like it is because mana steal would now be bounded by both health AND mana.  So on one hand it'll force players that use it to invest in health stat points.  Also, if will encourage players to be really creative.  For example, one secret in the game mechanics is that if you are standing on top of a health potion when you take damage, the potion will trigger immediately.  So you could use last will to pull a bunch of health potions to you, stand on them, stack mana steal and take over your max health while being healed to make yourself survive it.  It'll also pair well with blood curse, lots of possibilities.

Also you still have to use targeting combos to get it to pay off.

To prevent it from being too easy to get 120 mana, we could balance how much max mana enemies have.  Like if priests had only 40 mana it it cost them 20 to cast, you'd have to still target multiple to get the same 120 mana that you'd get from the current manasteal (6 enemies @ 20 mana stolen per enemy = 120 mana).

I think this does exactly what we want, you still can get 500+ mana if you're really clever but it's no longer as simple as targeting the whole map because the health cost scales as it stacks.

**Comment by SoulMuncher** [comment]
Yeah I think you're right. It feels a little strange that the spell doesn't scale with extra targeting in the usual sense,, but it checks all of the boxes we need to have checked, and it'll be much better than current mana steal. Let's try it out :)

120 per cast is too much though. 120 mana for 30hp!? You could kill a deathmason in one turn with that. That's beating the final boss at the cost of _half a health potion_, one cast of lesser heal, one shield, one stat point of hp, one sacrifice, one... Well, you get it. Removing enemy mana is also pretty strong on its own. So you're not just killing the deathmason but stopping him from summoning for another turn too. One spell, no combos needed, very powerful.

Honestly... I'm thinking it should be more in the realm of 60mana. What do you think?

Also, will this steal mana evenly from all targeted enemies (same behavior as send mana)? And if an enemy doesn't have enough to steal, I can split the remainder among the other targets so the player always gets 60 mana as long as there's 60 mana to be stolen.

**Comment by jdoleary** [comment]
> 120 per cast is too much though.

Yeah I see your point, but if it's only 60, you'd have to use a huge amount of health to get half-decent mana from it.  What if we lowered the per unit pull and made stacks increase the per unit pull? So like `steal 10 mana per unit per stack`?

The tradeoffs are tough here.  If we do a total cap it's too easy to pull a ton of mana from one unit, if we do a per unit cap it's too easy to get way too much mana from a massive targeting spell.
Or maybe there's a per unit cap _and_ a total cap.  So you can only take 20 mana per unit but there's a 100 max cap, so you can't just remove all of the deathmasons mana.  It'll require at least 5 targeted enemies to meet the cap?

> (same behavior as send mana)

yes.

**Comment by SoulMuncher** [comment]
A dual cap definitely solves balancing single target vs weak multi target. But that's what I did with the first balancing attempt and we felt kinda iffy about it. Even though it might be more balanced, I think it introduces other issues and nuances that make it a worse solution overall. I would vote for trying a total cap before trying this.

> if it's only 60, you'd have to use a huge amount of health to get half-decent mana from it.

I mean, if you asked me what a half decent amount of mana was, in number format, I would probably say 60. That's enough to easily triple the effectiveness of your next play. And even if you don't think 60 is good, you can get 120 with one stat point in hp or any amount of healing. That amount of mana is going to be way more valubale than 30 health most of the time, especially when you get to pick and choose when and whether or not you cast the spell.

I don't think you should get 200+ mana unless you've really invested into health or done some kind of cool sacrifice or last will combo like you mentioned, and at that point it should be easy to get several hundred. Anything beyond that isn't really measurable/doesn't have an effect on the game anyway.

I'm feeling pretty confident about 60 honestly. It's nothing compared to old mana steal, but I think it'll still be like... A if not S tier. Especially with health changes!

What do you think about trying 60 total cap and if it sucks and we can increase the amount or try the dual cap?

**Comment by jdoleary** [comment]
Okay, I'm on board! Let's go with 60 and we can always balance later.  I'm fairly confident about this balancing choice of doing a total cap, just uncertain on the specifics of the balancing.  But I think it'll definitely address the huge issue of mana steal being way too OP

**Review by jdoleary — CHANGES_REQUESTED** [review]
So one issue with this I see is that some enemies have huge mana relative to how much you can steal (priest has 160), so you could cast mana steal twice on the same priest and still get a max of 120 mana and you wouldn't even need to add targeting.  In fact you'd only benefit from targeting if you had already 3 stacks of mana steal (when targeting a priest).  I have to spend 206 health just to get 240 mana.  This feels not worth it.

I think we're headed in the right direction... but maybe we need to set the expense scaling of mana steal to 0.  That way you're always converting 1hp to 2 mana so long as the mana is available.

Also maybe we need to lower the max mana and cost to cast for units like priest?

File: src/cards/mana_steal.ts [hunk 1]
```diff
@@ -32,19 +32,30 @@ const spell: Spell = {
...
+      // Start with the amount we intend to steal, adjust after for loop
+      let totalManaStolen = base_mana_stolen * quantity;
+      let remainingManaToSteal = totalManaStolen;
+
+      // sort targets by current mana to carry remainder, in case you try to steal more mana than a unit has
+      targets.sort((a, b) => a.mana - b.mana);
+      for (let i = 0; i < targets.length; i++) {
+        const unit = targets[i];
+        if (unit) {
+          console.log(unit.mana);
```
[the remark this answers is not in the document]
Comment (SoulMuncher) [reply src/cards/mana_steal.ts:44]: Thank you. I should probably look over my changes a little closer too 😅

**Review by jdoleary — APPROVED** [review]

**Pull request merged**